### PR TITLE
feat: implement container authentication

### PR DESCRIPTION
**Pull request by pyrooka**
This PR implements the new container authentication flow, which is highly based on the IAM auth. This means a new token manager and authenticator have been created and covered with tests.

NOTE: this PR/branch is based on the IAM refactoring branch(shared-iam-related-code)/PR(#118), so SHOULD NOT be merged in before the other one.

### CODE REVIEW

**Comment by padamstx** [comment]
@pyrooka Looks like there's a linter error in the builds :)

File: ibm_cloud_sdk_core/token_managers/container_token_manager.py
Comment (pyrooka) [inline ibm_cloud_sdk_core/token_managers/container_token_manager.py:119]: I want to handle every error here, not just e.g. missing files, this is why I use the plain `Exception` here. If you think this is too common, I am open for suggestions.

Comment (padamstx) [reply ibm_cloud_sdk_core/token_managers/container_token_manager.py:119]: I think this is fine the way it is.  Not sure how much value there would be to creating a specific exception here (I assume that is what you were alluding to).

File: test/test_container_token_manager.py
Comment (padamstx) [inline test/test_container_token_manager.py:142]:
```suggestion
def test_retrieve_cr_token_fail():
```
?
Are you trying to test one of those "success fail" scenarios? :)

File: test/test_container_token_manager.py
Comment (padamstx) [inline test/test_container_token_manager.py:160]:
```suggestion
    access_token = token_manager.access_token
```
Using the name `cr_token` here is a bit confusing because you're actually assigning it from the "access_token" field which sort of implies that it's the IAM access token, and not the "cr token".

File: test/test_container_token_manager.py
Comment (padamstx) [inline test/test_container_token_manager.py:163]: Same comments about the name `cr_token` here and below.

File: test/test_utils.py
Comment (padamstx) [inline test/test_utils.py:300]: please add asserts to validate the rest of the properties that you are setting for this service in ibm-credentials-container.env.   I'm especially interested in the disable-ssl property since that is defined as a string in the config property, but the authenticator expects it to be a bool.

File: resources/ibm-credentials-container.env
Comment (padamstx) [inline resources/ibm-credentials-container.env:9]: Also, try a test with AUTH_DISABLE_SSL set explicitly to false (see my comments above re: the `__construct_authenticator` function).

**Review by padamstx — CHANGES_REQUESTED** [review]
Overall, these changes look good, but I found a problem related to our support of the AUTH_DISABLE_SSL config property that has most likely existed for quite some time.
I guess now's a good time to fix it :)

File: Authentication.md
Comment (pyrooka) [reply Authentication.md:207]: Sure! (I've copied this from the Node auth document, maybe we should modify that as well.)

File: ibm_cloud_sdk_core/authenticators/container_authenticator.py
Comment (pyrooka) [reply ibm_cloud_sdk_core/authenticators/container_authenticator.py:55]: Nice! It's a good time to fix a 2 years old typo: https://github.com/IBM/python-sdk-core/blob/main/ibm_cloud_sdk_core/authenticators/iam_authenticator.py#L51 😄 (I should pay more attention when copy-paste...)

File: test/test_container_token_manager.py
Comment (pyrooka) [reply test/test_container_token_manager.py:142]: Yeah, I am trying to test scenarios that don't even exist. (Just kidding :) )

File: test/test_container_token_manager.py
Comment (pyrooka) [reply test/test_container_token_manager.py:160]: Yep, you are totally right.

File: Authentication.md
Comment (dpopp07) [reply Authentication.md:207]: Good to know, I'll double check what I did 🙂

Comment (dpopp07) [reply Authentication.md:207]: I updated my changes to this file in the Node core to match Go, I just can't push them at the moment

File: ibm_cloud_sdk_core/token_managers/container_token_manager.py
Comment (dpopp07) [inline ibm_cloud_sdk_core/token_managers/container_token_manager.py:31]: Should this be `cr_token_filename` ?

**Review by dpopp07 — COMMENTED** [review]
Looks good once Phil's comments are addressed (I'll wait until then to officially approve)

I just found what I think are a few typos

File: ibm_cloud_sdk_core/authenticators/container_authenticator.py
Comment (pyrooka) [reply ibm_cloud_sdk_core/authenticators/container_authenticator.py:36]: Good catch!

File: ibm_cloud_sdk_core/token_managers/container_token_manager.py
Comment (pyrooka) [reply ibm_cloud_sdk_core/token_managers/container_token_manager.py]: Oops, yes!

**Review by padamstx — CHANGES_REQUESTED** [review]
Getting pretty close, but a few small changes needed.
Plus I'm sneaking in a request to support "AUTHTYPE" as an alias for the "AUTH_TYPE" config property :)

**Review by padamstx — APPROVED** [review]
I've pushed a commit to address the changes I requested in my previous review.  So I approved of those changes :)

**Comment by pyrooka** [comment]
Thank you for fixing these issues @padamstx !

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Pull request merged**
